src_c/geometry.c

```diff
@@ -8,6 +8,150 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+/*
+ * origin, direction, max_dist
+ * origin, angle, max_dist
+ * line
+ *
+ * sets the error messages
+ * 1 if success
+ * o if it fails
+ */
+static int
+_pg_extract_ray_from_object_fastcall(PyObject *const *args, Py_ssize_t nargs,
+                                     pgLineBase *line, double *max_t)
+{
+    if (nargs == 1) {
+        if (!pgLine_FromObject(args[0], line)) {
+            PyErr_SetString(
+                PyExc_TypeError,
+                "line parameter must be a Line or a LineLike object");
+            return 0;
+        }
+
+        *max_t = 1.0;
+
+        return 1;
+    }
+    else if (nargs == 3) {
+        if (!pg_TwoDoublesFromObj(args[0], &line->x1, &line->y1)) {
+            PyErr_SetString(
+                PyExc_TypeError,
+                "Invalid ray origin value, must be a pair of numeric values");
+            return 0;
+        }
+
+        if (PyNumber_Check(args[1])) {
+            double angle;
+            if (!pg_DoubleFromObj(args[1], &angle)) {
+                PyErr_SetString(PyExc_TypeError,
+                                "Invalid ray angle value, must be numeric");
+                return 0;
+            }
+
+            line->x2 = line->x1 - cos(angle * PI / 180);
+            line->y2 = line->y1 - sin(angle * PI / 180);
+        }
+        else if (!pg_TwoDoublesFromObj(args[1], &line->x2, &line->y2)) {
+            PyErr_SetString(PyExc_TypeError,
+                            "expected a pair of floats or a single float");
+            return 0;
```

Comment:
You don't expect a single float anymore, so you can change the error message now.